src/app/mat-select-search/mat-select-search.component.ts

```diff
@@ -499,40 +493,14 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, ControlValue
       const ariaActiveDescendantId = this.matSelect._getAriaActiveDescendant();
       const index = this._options.toArray().findIndex(item => item.id === ariaActiveDescendantId);
       if (index !== -1) {
-        const activeDescendant = this._options.toArray()[index];
-        this.liveAnnouncer.announce(
-          activeDescendant.viewValue + ' '
-          + this.getAriaIndex(index)
-          + this.indexAndLengthScreenReaderText
-          + this.getAriaLength()
-        );
+        this.activeDescendant?.removeAttribute('aria-selected');
+        this.activeDescendant = this._options.toArray()[index]._getHostElement();
+        this.activeDescendant.setAttribute('aria-selected', 'true');
```

Comment:
In theory, there could be events other than keydown/up that change the active descendant, so this and
The next line would be outdated. Did you investigate if this can happen?
Maybe there is an Observable for when the active descendant changes?

Reply:
The only situation I could figure out was that the set attributes become outdated after the component gets blurred. Therefore I introduced a `unselectActiveDescendant` method that gets called everytime the `blur` method gets called, so that the `aria-selected` and the `aria-activedescendant` attributes are removed properly.